src/libsync/propagateremotemkdir.cpp

```diff
@@ -28,6 +28,46 @@ namespace OCC {
 
 Q_LOGGING_CATEGORY(lcPropagateRemoteMkdir, "nextcloud.sync.propagator.remotemkdir", QtInfoMsg)
 
+PropagateRemoteMkdir::PropagateRemoteMkdir(OwncloudPropagator *propagator, const SyncFileItemPtr &item)
+    : PropagateItemJob(propagator, item)
+    , _deleteExisting(false)
+    , _uploadEncryptedHelper(nullptr)
+    , _parallelism(FullParallelism)
+{
+    const auto rootPath = [=]() {
+        const auto result = propagator->_remoteFolder;
+        if (result.startsWith('/')) {
+            return result.mid(1);
+        } else {
+            return result;
+        }
+    }();
+    const auto path = _item->_file;
+    const auto slashPosition = path.lastIndexOf('/');
+    const auto parentPath = slashPosition >= 0 ? path.left(slashPosition) : QString();
+
+    SyncJournalFileRecord parentRec;
+    bool ok = propagator->_journal->getFileRecord(parentPath, &parentRec);
+    if (!ok) {
+        done(SyncFileItem::NormalError);
```

Comment:
IMO the current set of ErrorTypes is not appropriate/enough anymore, having more descriptive ones here would help a lot when getting logs from issue reporters. This bugged me already while debugging another issue a while ago. But just leaving that comment here, pretty unrelated, maybe this makes it to a note if I'm not the only one feeling that way.

Reply:
You're right, I was thinking the same the past few weeks. Will need to be carefully planned though this is quite central to the logic.